juju/juju
# Remove the componentisation of resources and payloads

This looks bigger than it is - a lot of code is moved around. The resources and payloads infrastructure is moved from the so called "component based approach" to the standard juju model. The basis of the change is that (most of the) code from the "components" package is moved to the uniter worker and remodelled as part of the hook context struct. The hook commands are also moved to live with all the others. The manifold used to make the uniter worker sets up the payload and resource facades and passes these into the runner context factory. From there hook commands have access to the necessary api calls.

Where relevant, a lot of the tests were rewritten to use gomock instead of the hand crafted stubs, also adding somewhat to the bulk of the PR.

## QA steps

See QA steps for these previous PRs

https://github.com/juju/juju/pull/14058
https://github.com/juju/juju/pull/14049

Essentially, deploy charms with payloads and resources and run various hook commands.
